src/PacketHeader.cpp

```diff
@@ -92,27 +94,29 @@ uint64_t PacketHeader::usecTime()
 DefaultHeader::DefaultHeader(JackTrip* jacktrip)
     : PacketHeader(jacktrip), mJackTrip(jacktrip)
 {
-    mHeader.TimeStamp     = 0;
-    mHeader.SeqNumber     = 0;
-    mHeader.BufferSize    = 0;
-    mHeader.SamplingRate  = 0;
-    mHeader.BitResolution = 0;
-    //mHeader.NumInChannels = 0;
-    //mHeader.NumOutChannels = 0;
-    mHeader.NumChannels    = 0;
-    mHeader.ConnectionMode = 0;
+    mHeader.TimeStamp                  = 0;
+    mHeader.SeqNumber                  = 0;
+    mHeader.BufferSize                 = 0;
+    mHeader.SamplingRate               = 0;
+    mHeader.BitResolution              = 0;
+    mHeader.NumIncomingChannelsFromNet = 0;
+    mHeader.NumOutgoingChannelsToNet   = 0;
 }
 
 //***********************************************************************
 void DefaultHeader::fillHeaderCommonFromAudio()
 {
-    mHeader.TimeStamp      = PacketHeader::usecTime();
-    mHeader.BufferSize     = mJackTrip->getBufferSizeInSamples();
-    mHeader.SamplingRate   = mJackTrip->getSampleRateType();
-    mHeader.BitResolution  = mJackTrip->getAudioBitResolution();
-    mHeader.NumChannels    = mJackTrip->getNumChannels();
-    mHeader.ConnectionMode = static_cast<int>(mJackTrip->getConnectionMode());
-    //printHeader();
+    mHeader.TimeStamp                  = PacketHeader::usecTime();
+    mHeader.BufferSize                 = mJackTrip->getBufferSizeInSamples();
+    mHeader.SamplingRate               = mJackTrip->getSampleRateType();
+    mHeader.BitResolution              = mJackTrip->getAudioBitResolution();
+    mHeader.NumIncomingChannelsFromNet = mJackTrip->getNumOutputChannels();
+
+    if (0 == mJackTrip->getNumInputChannels()) {
+        mHeader.NumOutgoingChannelsToNet = std::numeric_limits<uint8_t>::max();
```

Comment:
If the legacy client sends 0 as outgoing number of channels we have to do the same.